packages/flutter/lib/src/material/segmented_button.dart

```diff
@@ -436,6 +444,16 @@ class SegmentedButton<T> extends StatefulWidget {
 @visibleForTesting
 class SegmentedButtonState<T> extends State<SegmentedButton<T>> {
   bool get _enabled => widget.onSelectionChanged != null;
+  bool _hovering = false;
+  bool _focused = false;
+  bool get _selected => widget.selected.contains(widget.segments.first.value);
+
+  Set<WidgetState> get _states => <WidgetState>{
+    if (!_enabled) WidgetState.disabled,
+    if (_hovering) WidgetState.hovered,
+    if (_focused) WidgetState.focused,
+    if (_selected) WidgetState.selected,
```

Comment:
Can we also add a pressed state?

Reply:
I tried, but it looks like retrieving the pressed state from `ButtonStyleButton` in `InkWell` is tricky without creating listeners for each segment of the `statesController`. A better way to do this would be to add a selected callback that returns a `bool` on the buttons, similar to a checkbox or switch. If you have an easier way to do this, I would be curious to know. :)